### PR TITLE
feat: [CDS-78721]: support terragrunt running in parallel having uniq…

### PR DESCRIPTION
**Pull request by vlica-harness**
…ue baseDir
The purpose of this PR is to support terragrunt steps with the same provisionerId to run in parallel.
Previously this was possible only when provisionerId is different for all same for examople: terragrunt plan steps.
Because the BaseDir was created based on `"./terragrunt-working-dir/${ACCOUNT_ID}/${ENTITY_ID}";`
ENTITY_ID is created based on accountId + orgId + projectId + provisionerId.

Now, BaseDir will have one more folder which is going to be created randomly and assigned to UNIQUE_DIRECTORY

`"./terragrunt-working-dir/${ACCOUNT_ID}/${ENTITY_ID}/${UNIQUE_DIRECTORY}";`

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52541)

### CODE REVIEW

**Comment by vlica-harness** [comment]
trigger commonchecks

**Comment by vlica-harness** [comment]
trigger uts_JDK11

**Comment by vlica-harness** [comment]
trigger uts_JDK11

**Comment by vlica-harness** [comment]
trigger uts_JDK11

**Comment by vlica-harness** [comment]
trigger ut3

**Comment by vlica-harness** [comment]
trigger ut3

**Comment by vlica-harness** [comment]
trigger uts_JDK11

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/terragrunt/TerragruntDestroyTaskNG.java [hunk 1]
```diff
@@ -90,8 +90,15 @@ public DelegateResponseData run(TaskParameters parameters) throws IOException, J
     LogCallback destroyLogCallback =
         taskService.getLogCallback(getLogStreamingTaskClient(), DESTROY, commandUnitsProgress);
 
-    String baseDir =
-        TerragruntTaskService.getBaseDir(destroyTaskParameters.getAccountId(), destroyTaskParameters.getEntityId());
+    String baseDir;
+    if (destroyTaskParameters.isUseUniqueDirectoryForBaseDir()) {
```
Comment (sourabh-awashti) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/terragrunt/TerragruntDestroyTaskNG.java:94]: why are we using static instance here?? is it serving any purpose?

**Review by tmacari-harness — DISMISSED** [review]
> Reviewed 17 of 17 files at r1, 4 of 4 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52541)* status: 22 file(s) to review. 1 of 2 approvals obtained.

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/terragrunt/TerragruntDestroyTaskNG.java (hunk 1 above)
Comment (vlica-harness) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/terragrunt/TerragruntDestroyTaskNG.java:94]: It's used just as a kind of utility method. Just got the same for that was used previously

**Review by sourabh-awashti — APPROVED** [review]

**Comment by vlica-harness** [comment]
trigger uts_JDK11

**Comment by vlica-harness** [comment]
trigger uts_JDK11

**Review by tmacari-harness — APPROVED** [review]

**Comment by vlica-harness** [comment]
trigger review

**Pull request merged**